graph/src/data/graphql/effort.rs

```diff
@@ -276,12 +292,19 @@ impl LoadManager {
             jailed_queries: RwLock::new(HashSet::new()),
             kill_state: RwLock::new(KillState::new()),
             effort_gauge,
+            query_counters,
         }
     }
 
-    pub fn add_query(&self, shape_hash: u64, duration: Duration) {
+    /// Record that we spent `duration` amount of work for the query
+    /// `shape_hash`, where `cache_status` indicates whether the query
+    /// was cached or had to actually run
+    pub fn record_work(&self, shape_hash: u64, duration: Duration, cache_status: CacheStatus) {
         if !*LOAD_MANAGEMENT_DISABLED {
             self.effort.add(shape_hash, duration, &self.effort_gauge);
+            self.query_counters
```

Comment:
I feel like this logging should be irrespective of LOAD_MANAGEMENT_DISABLED which is for the feature that rejects queries when under load. I think you would want to use the counters regardless of whether we turn that off. So, just move outside the `if` block.

Reply:
Good point. Made the counting unconditional.